### PR TITLE
snap package.

### PR DESCRIPTION
**Pull request by mikeroyal**
Feature Request: snap package

Includes a wrapper script and snapcraft.yaml to build [snap](https://snapcraft.io/). I placed the snap here because the current build supports Debian/Ubuntu.

Signed-off-by: Michael Royal <mikerr90@outlook.com>

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

File: debian/snapcraft/snap/snapcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+      apt -y upgrade
+
+  ceph:
+    after:
+      - gnome
+    plugin: cmake
+    source:
+      # Ceph is only available for amd64 (x86_64),
+      # so fail the build on other architectures.
+      - on amd64: https://download.ceph.com/debian-luminous/pool/main/c/ceph/ceph_12.2.12ubuntu_x86_64.deb
```
Comment (sebastian-philipp) [inline debian/snapcraft/snap/snapcraft.yaml]: Hm. :-1: for adding a link to a luminous release.

File: debian/snapcraft/snap/snapcraft.yaml [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+architectures:
+  - build-on: amd64
+
+parts:
+  gnome:
+    plugin: nil
+    build-packages:
+      - software-properties-common
+    override-pull: |
+      add-apt-repository -y ppa:ubuntu-desktop/gnome-3-26
```
Comment (sebastian-philipp) [inline debian/snapcraft/snap/snapcraft.yaml]: ?

File: debian/snapcraft/snap/snapcraft.yaml [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+      DEB_URL=$(curl -w "%{url_effective}\n" -I -L -s -S "${DEB_API}" -o /dev/null)
+      VERSION=$(echo "${DEB_URL}" | cut -d'_' -f2)
+      echo $VERSION > $SNAPCRAFT_STAGE/version
+
+      snapcraftctl build
+    build-packages:
+      - curl
+      - dpkg
+      - sed
+    stage-packages:
```
Comment (sebastian-philipp) [inline debian/snapcraft/snap/snapcraft.yaml]: where does this list come from?

**Comment by mikeroyal** [comment]
Hi @sebastian-philipp, I will make the changes you suggested and put this in a different repo.

**Pull request closed without merging**